src/main/java/net/machpi/runelite/influxdb/activity/GameEvent.java

```diff
@@ -0,0 +1,424 @@
+    MG_BURTHORPE_GAMES_ROOM("Burthorpe Games Room", LocationType.MINIGAMES, 8781),
+    MG_CASTLE_WARS("Castle Wars", LocationType.MINIGAMES, 9520),
+    MG_CLAN_WARS("Clan Wars", LocationType.MINIGAMES, 13135, 13134, 13133, 13131, 13130, 13387, 13386),
+    MG_DUEL_ARENA("Duel Arena", LocationType.MINIGAMES, 13362),
+    MG_FISHING_TRAWLER("Fishing Trawler", LocationType.MINIGAMES, 7499),
+    MG_GAUNTLET("Gauntlet", LocationType.MINIGAMES, 12995),
+    MG_INFERNO("The Inferno", LocationType.MINIGAMES, 9043),
+    MG_LAST_MAN_STANDING("Last Man Standing", LocationType.MINIGAMES, 13660, 13659, 13658, 13916, 13915, 13914),
+    MG_MAGE_TRAINING_ARENA("Mage Training Arena", LocationType.MINIGAMES, 13462, 13463),
+    MG_NIGHTMARE_ZONE("Nightmare Zone", LocationType.MINIGAMES, 9033),
+    MG_PEST_CONTROL("Pest Control", LocationType.MINIGAMES, 10536),
+    MG_PYRAMID_PLUNDER("Pyramid Plunder", LocationType.MINIGAMES, 7749),
+    MG_ROGUES_DEN("Rogues' Den", LocationType.MINIGAMES, 11855, 11854, 12111, 12110),
+    MG_SORCERESS_GARDEN("Sorceress's Garden", LocationType.MINIGAMES, 11605),
+    MG_TEMPLE_TREKKING("Temple Trekking", LocationType.MINIGAMES, 8014, 8270, 8256, 8782, 9038, 9294, 9550, 9806),
+    MG_TITHE_FARM("Tithe Farm", LocationType.MINIGAMES, 6968),
+    MG_TROUBLE_BREWING("Trouble Brewing", LocationType.MINIGAMES, 15150),
+    MG_TZHAAR_FIGHT_CAVES("Tzhaar Fight Caves", LocationType.MINIGAMES, 9551),
+    MG_TZHAAR_FIGHT_PITS("Tzhaar Fight Pits", LocationType.MINIGAMES, 9552),
+    MG_VOLCANIC_MINE("Volcanic Mine", LocationType.MINIGAMES, 15263, 15262),
+
+    // Raids
+    RAIDS_CHAMBERS_OF_XERIC("Chambers of Xeric", LocationType.RAIDS, Varbits.IN_RAID),
+    RAIDS_THEATRE_OF_BLOOD("Theatre of Blood", LocationType.RAIDS, Varbits.THEATRE_OF_BLOOD),
+
+    POI_FISHING_GUILD("Fishing Guild", LocationType.POI, 10293),
+    POI_OTTOS_GROTTO("Otto's Grotto", LocationType.POI, 10038),
+    POI_PLAYER_OWNED_HOUSE("Player Owned House", LocationType.POI, 7769),
+    ;
+
+    private static final Map<Integer, GameEvent> FROM_REGION;
+    private static final List<GameEvent> FROM_VARBITS;
+
+    static {
+        ImmutableMap.Builder<Integer, GameEvent> regionMapBuilder = new ImmutableMap.Builder<>();
+        ImmutableList.Builder<GameEvent> fromVarbitsBuilder = ImmutableList.builder();
+        for (GameEvent gameEvent : GameEvent.values()) {
+            if (gameEvent.getVarbits() != null) {
+                fromVarbitsBuilder.add(gameEvent);
+                continue;
+            }
+
+            if (gameEvent.getRegionIds() == null) {
+                continue;
+            }
+
+            for (int region : gameEvent.getRegionIds()) {
+                regionMapBuilder.put(region, gameEvent);
+            }
+        }
+        FROM_REGION = regionMapBuilder.build();
+        FROM_VARBITS = fromVarbitsBuilder.build();
+    }
+
+    @Nullable
+    private String location;
```

Comment:
make things final that can be final.

Reply:
The fields that can be final are already final. `location` can't be final in this case.